packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.service.js

```diff
@@ -53,4 +55,77 @@ export default class SofpthoneService {
       .get(`/telephony/${billingAccount}/line/${serviceName}/devices`)
       .then(({ data }) => data);
   }
+
+  getThemes() {
+    return this.$http.get('/telephony/softphone/themes').then(({ data }) => {
+      return this.$q.all(
+        data.map((theme) =>
+          this.$http
+            .get(`/telephony/softphone/themes/${theme}`)
+            .then((res) => res.data),
+        ),
+      );
+    });
```

Comment:
Here it need to use iceberg call for to have the expend in only one call http.

```suggestion
    return this.iceberg('/telephony/softphone/themes')
      .query()
      .expand('CachedObjectList-Pages')
      .execute()
      .$promise.then(({ data }) => data)
```